firstvoices/backend/serializers/import_job_serializers.py

```diff
@@ -0,0 +1,136 @@
+        validators=[SupportedFileType(mimetypes=["text/csv", "text/plain"])],
+    )
+    run_as_user = serializers.CharField(required=False)
+    validation_result = ImportReportSerializer(read_only=True)
+
+    class Meta:
+        model = ImportJob
+        fields = SiteContentLinkedTitleSerializer.Meta.fields + (
+            "mode",
+            "validation_result",
+            "run_as_user",
+            "data",
+        )
+
+    def create_file(self, file_data, filetype, site):
+        user = self.context["request"].user
+        file = filetype(
+            content=file_data,
+            site=site,
+            created_by=user,
+            last_modified_by=user,
+        )
+        file.save()
+        return file
+
+    def create(self, validated_data):
+        validated_data["site"] = get_site_from_context(self)
+        file = self.create_file(validated_data["data"], File, validated_data["site"])
+
+        try:
+            table = tablib.Dataset().load(
+                file.content.read().decode("utf-8-sig"), format="csv"
+            )
+
+            # Validate headers
+            # If required headers not present, raise ValidationError
+            check_required_headers(table.headers)
+
+            # else, print warnings for extra or invalid headers
+            validate_headers(table.headers)
+
+            # If the file is valid, create an ImportJob instance and save the file
+            title = validated_data.get("title", "")
+            mode = validated_data.get("mode", None)
+            run_as_user = validated_data.get("run_as_user", None)
+
+            entry = ImportJob(
+                title=title,
+                data=file,
+                site=validated_data["site"],
+            )
+            if mode:
+                entry.mode = mode
+
+            # Validate the user and then attach the foreign user object
+            # and check if the user requesting is a superadmin
+            if run_as_user:
+                if not is_superadmin(self.context["request"].user, None):
+                    # Only superadmins can use this field
+                    raise serializers.ValidationError(
+                        detail={
+                            "run_as_user": [
+                                "This field can only be used by superadmins."
+                            ]
+                        }
+                    )
+                user_model = get_user_model()
+                username_field = user_model.USERNAME_FIELD
+                user = user_model.objects.filter(**{username_field: run_as_user})
+                if len(user) == 0:
+                    raise serializers.ValidationError(
+                        detail={
+                            "run_as_user": [
+                                f"User with the provided {username_field} not found."
+                            ]
+                        }
+                    )
+                entry.run_as_user = user[0]
```

Comment:
I think you could extract the logic for getting a valid username into its own function, and since there is a possibility of validation errors it should happen before we start creating the ImportJob model